internal/controller/flow_falco.go

```diff
@@ -0,0 +1,160 @@
+func NewFalcoEventHandler(eventChan chan<- string) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		var body struct {
+			Output string `json:"output"`
+		}
+
+		if err := json.NewDecoder(r.Body).Decode(&body); err != nil {
+			http.Error(w, err.Error(), http.StatusBadRequest)
+			return
+		}
+		eventChan <- body.Output
+		w.WriteHeader(http.StatusOK)
+	}
+}
+
+// filterIllumioTraffic filters out events related to Illumio network traffic.
+func filterIllumioTraffic(body string) bool {
+	return strings.Contains(body, "illumio_network_traffic")
+}
+
+func convertFalcoEventToFlow(event FalcoEvent) (*pb.FalcoFlow, error) {
+	layer3Message, err := createLayer3Message(event.SrcIP, event.SrcIP, event.IpVersion)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create Layer3 message falco flows: %v", err)
+	}
+
+	srcPort, err := strconv.ParseUint(event.SrcPort, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("invalid source port: %v", err)
+	}
+
+	dstPort, err := strconv.ParseUint(event.DstPort, 10, 32)
+	if err != nil {
+		return nil, fmt.Errorf("invalid destination port: %v", err)
+	}
+
+	layer4Message, err := CreateLayer4Message(event.Proto, uint32(srcPort), uint32(dstPort), event.IpVersion)
+	if err != nil {
+		return nil, fmt.Errorf("could not create Layer4 Message for Falco flow %v", err)
+	}
+
+	flow := &pb.FalcoFlow{
+		Layer3: layer3Message,
+		Layer4: layer4Message,
+	}
+
+	return flow, nil
+}
+
+func createLayer3Message(source string, destination string, ipVersion string) (*pb.IP, error) {
+	if ipVersion == "ipv4" {
+		return &pb.IP{Source: source, Destination: destination, IpVersion: pb.IPVersion_IP_VERSION_IPV4}, nil
+	} else if ipVersion == "ipv6" {
+		return &pb.IP{Source: source, Destination: destination, IpVersion: pb.IPVersion_IP_VERSION_IPV6}, nil
+	}
+	return &pb.IP{Source: source, Destination: destination, IpVersion: pb.IPVersion_IP_VERSION_IP_NOT_USED_UNSPECIFIED}, nil
+}
+
+// CreateLayer4Message converts event protocol and ports to a Layer4 proto message
+func CreateLayer4Message(proto string, srcPort, dstPort uint32, ipVersion string) (*pb.Layer4, error) {
+	switch proto {
+	case "tcp":
+		return &pb.Layer4{
+			Protocol: &pb.Layer4_Tcp{
+				Tcp: &pb.TCP{
+					SourcePort:      srcPort,
+					DestinationPort: dstPort,
+					Flags:           &pb.TCPFlags{},
+				},
+			},
+		}, nil
+	case "udp":
+		return &pb.Layer4{
+			Protocol: &pb.Layer4_Udp{
+				Udp: &pb.UDP{
+					SourcePort:      srcPort,
+					DestinationPort: dstPort,
+				},
+			},
+		}, nil
+	case "icmp":
+		if ipVersion == "ipv4" {
+			return &pb.Layer4{
+				Protocol: &pb.Layer4_Icmpv4{
+					Icmpv4: &pb.ICMPv4{},
+				},
+			}, nil
+		} else if ipVersion == "ipv6" {
+			return &pb.Layer4{
+				Protocol: &pb.Layer4_Icmpv6{
+					Icmpv6: &pb.ICMPv6{},
+				},
+			}, nil
+		}
+	default:
+		return nil, fmt.Errorf("unknown protocol: %s", proto)
```

Comment:
I'd rather silently ignore. Don't return an error for that. Because it seems like we'lll also get UDS flows, RAW socket flows, etc.?
https://github.com/falcosecurity/libs/blob/230ddfb91d075241391697900ab4281aa67fe108/userspace/libsinsp/parsers.cpp#L2564-L2565
We should just ignore those.